src/Documentation/resources/Best_Practices.md

```diff
@@ -0,0 +1,422 @@
+# Best Practices
+
+Orleans was designed in 2014 with the intention of removing low-level
+abstractions from actor model frameworks. Grains (actors stored in memory) are
+used to maintain the lifecycle among other features. Orleans is based on the
+actor model and fits some projects better that others.
+
+## Orleans should be considered when:
+
+-   Significant number (hundreds to millions) of loosely coupled entities
```

Comment:
I'm not sure how best to reflect the fact that there are two number here really - total number of entities, which can easily be in the billions and trillions, and the subset of that total number that is activated at any point in time.

User profiles are a good example. It is easy to have a grain for every person on Earth within a small cluster, so long as only a subset of users are active at any point in time.

Reply:
Corrected